### ISSUE TITLE
Loading polmineR twice may fail in examples

### ISSUE THREAD

**Issue by ablaette** [PolMine/polmineR#164]
When picking up the development of my fulltext package, I realized that relying on polmineR and the corpora shipped with the package may fail if multiple examples call `library(polmineR)` and `use("polmineR")`. This behavior is clearly unintended Note that it occurrs only if the environment variable CORPUS_REGISTRY is not set. 

See the following minimal example.

```r
Sys.setenv("CORPUS_REGISTRY" = "")

pkgname <- "annolite"
source(file.path(R.home("share"), "R", "examples-header.R"))
base::assign(".oldSearch", base::search(), pos = 'CheckExEnv')
base::assign(".old_wd", base::getwd(), pos = 'CheckExEnv')

library(polmineR)
use("polmineR")

cleanEx()

library(polmineR)
use("polmineR")
```

**Comment by ablaette** [comment]
I would like to follow up on this issue by elaborating the example somewhat. 

To understand the problem, I have extracted the essence of the `cleanEx()` function that is called when running examples: There is a call of `tools::detachPackages()` which will trigger calling polmineR's  `.onDetach()` function which will delete temporary files.

```r
Sys.setenv("CORPUS_REGISTRY" = "")

pkgname <- "annolite"
source(file.path(R.home("share"), "R", "examples-header.R")) # loads cleanEx
base::assign(".oldSearch", base::search(), pos = 'CheckExEnv')

library(polmineR)
use("polmineR")

sch <- search()
newitems <- sch[! sch %in% .oldSearch]
tools:::detachPackages(newitems)
```

A this stage,  polmineR's temporaryfiles are removed.

```r
list.files(tempdir())
```

But the package, though it is not attached any more,  is still loaded:

```r
isNamespaceLoaded("polmineR")
```

As a consequence, when we load polmineR again ... 

```r
library(polmineR)
```

The package is re-attached but not loaded, because it is still loaded. The result is, that `.onLoad()` is not called and the temproary directories are not created.

```r
list.files(tempdir())
```

My conclusion would be that `.onDetach()` should mirror what `.onAttach()` does and `.onLoad()` should have corresponding actions in `.onUnload()`. In other words: Temporary files are created in `.onLoad()` and should be removed in `.onUnload()`, and not in `.onDetach()`.

**Comment by ablaette** [comment]
This is addressed in polmineR v0.8.5.9007: The temporary directory structure is removed in .onUnload() rather than .onDetach() now. Taking the examples in [my annolite package](https://github.com/PolMine/annolite) as a check, this solution works as intended.
